modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -15,12 +18,42 @@ const fetchTextData = async (url: string, errMsg: string) => {
   return res.text();
 };
 
+const createFetchGitHash = () => {
+  let gitHashCache: string;
+  return {
+    fetchGitHash: async () => {
+      if (gitHashCache) return gitHashCache;
+      try {
+        const gitHash = await fetchTextData(GIT_HASH_URL, 'Could not find current version or git hash');
+        gitHashCache = gitHash;
+        return gitHash;
+      } catch (e) {
+        throw new Error('Unsuccessful git hash fetch');
+      }
+    },
+    resetGitHashCache: () => {
+      gitHashCache = '';
+    },
+  };
+};
+
+const { fetchGitHash, resetGitHashCache } = createFetchGitHash();
+
 interface AtlasSpecUrlParams {
   apiKeyword: string;
   apiVersion?: string;
   resourceVersion?: string;
 }
 
+const ensureSavedVersionDataMatches = async (versions: VersionData, apiVersion?: string, resourceVersion?: string) => {
+  // Check that requested versions are included in saved version data
+  if (apiVersion) {
+    if (!versions.major.includes(apiVersion) || (resourceVersion && !versions[apiVersion].includes(resourceVersion))) {
+      throw new Error('Last successful saved build data does not include necessary version data');
```

Comment:
I think it could be helpful to include the api version and resource version info in the logs as a way to debug in case of an error